system/Format/JSONFormatter.php

```diff
@@ -56,13 +56,13 @@ class JSONFormatter implements FormatterInterface
 	 */
 	public function format($data)
 	{
-		$options = JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES;
+		$options = JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES | JSON_PARTIAL_OUTPUT_ON_ERROR;
 
 		$options = ENVIRONMENT === 'production' ? $options : $options | JSON_PRETTY_PRINT;
 
 		$result = json_encode($data, $options, 512);
 
-		if (json_last_error() !== JSON_ERROR_NONE)
+		if ( ! $result && json_last_error() !== JSON_ERROR_NONE)
```

Comment:
I think this is a condition we want to have here. This should also fix the tests.
```suggestion
		if (! in_array(json_last_error(), [JSON_ERROR_NONE, JSON_ERROR_RECURSION]))
```